client/client.ts

```diff
@@ -164,7 +178,9 @@ export class Client {
         this.connectPacket.clean = false;
         this.ctx.close();
       } catch (err) {
+        // TODO: can we replace this with an assert that ensures err to be instanceof Error?
```

Comment:
@seriousme question for you: if we can safely assume `err` to be an instance of `Error`, can we simply cast it into an `Error` at compile-time or assert it to be an `Error` at run time?

Reply:
I checked, we currently can't assume `err` to be an instance:
https://github.com/seriousme/opifex/blob/0430e16fb68a94f942f5078d806344bed216c0d3/client/client.ts#L131

And similarly:
https://github.com/seriousme/opifex/blob/0430e16fb68a94f942f5078d806344bed216c0d3/node/tcpClient.ts#L99

and:
https://github.com/seriousme/opifex/blob/0430e16fb68a94f942f5078d806344bed216c0d3/deno/tcpClient.ts#L73

But we could change all 3 of  them to produce an Error object. We just need to make sure then that in `bin/mqtt.js` we create a nice error message.

Reply:
That seems to be already the case: 

https://github.com/seriousme/opifex/blob/0430e16fb68a94f942f5078d806344bed216c0d3/bin/mqtt.ts#L211

I've wrapped the strings into Error objects in those three lines and added assertions to make sure caught values are instanceof errors.